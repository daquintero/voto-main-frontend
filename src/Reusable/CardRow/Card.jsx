// Absolute Imports
import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ instance }) => (
  <div>
    {instance.title}
  </div>
);

Card.propTypes = {
  instance: PropTypes.instanceOf(Object).isRequired,
};

export default Card;
