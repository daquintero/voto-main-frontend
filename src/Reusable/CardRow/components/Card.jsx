// Absolute Imports
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import moment from 'moment';

// Functions
import imageUrl from '../../../shared/utils/imageUrl';


const Card = ({ instance, index }) => {
  const image = instance.media.images[0];
  const primary = index === 0;
  return (
    <Link to="/" className="card-row__card">
      <img
        src={imageUrl(image.url)}
        alt={image.title}
        className={classNames({ primary })}
      />
      <div className="card-row__card__info">
        {primary ? (
          <h3>{instance.title}</h3>
        ) : (
          <h4>{instance.title}</h4>
        )}
        <span className="card-row__card__time">{moment(instance.dateCreated).fromNow()}</span>
      </div>
    </Link>
  );
};

Card.propTypes = {
  instance: PropTypes.instanceOf(Object).isRequired,
  index: PropTypes.number.isRequired,
};

export default Card;
