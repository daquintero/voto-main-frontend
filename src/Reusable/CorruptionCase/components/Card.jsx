// Generic Corruption Case Card
import React, { PureComponent } from 'react';
import { Card, CardBody, Row, Col } from 'reactstrap';
import PropTypes from 'prop-types';
// Routing for each card
import { Link } from 'react-router-dom';
// Numerical Formatting
import numeral from 'numeral';
// Date conversion
import moment from 'moment';
// Date language
import 'moment/locale/es';
// Layout Map
import layoutMap from '../styling/layoutMap';
// Configure moment settings
moment.locale('es');

class CorruptionCaseCard extends PureComponent {
  static propTypes = {
    instance: PropTypes.instanceOf(Object).isRequired,
  };

  render() {
    const { instance } = this.props;

    const statisticsItem = stats => (
      <div className="p-2 text-center justify-content-center" key={stats.id}>
        <i className={`fal p-2 enlarge fa-${stats.icon}`} aria-hidden="true" />
        <div className="justify-content-between">
          <p className="small m-0" >
            {/* Check if number or string */}
            {(typeof stats.data === 'number') ?
              numeral(stats.data).format('0a')
              : stats.data}
          </p>
          <p className="small m-0" > {stats.title} </p>
        </div>
      </div>
    );

    const imgFalseCard = instance => (
      <Card className="bg-light border-0 rounded-0 small-enlarge mincontent">
        <Link to={`/corrupcion-caso/${instance.id}`}>
          <Row noGutters className="h-100">
            <Col
              xs={layoutMap[instance.size].xs.inner.imgWrapper}
              md={layoutMap[instance.size].md.inner.imgWrapper}
            >
              <CardBody className="p-2 justify-self-center">
                <h4 className="card-title mb-0">{instance.title}</h4>
              </CardBody>
              <div className="d-flex justify-content-between bg-eb">
                {/* If there are three or more statistics,
                only display limits of 3 or map less than that */}
                {instance.statistics && instance.statistics[3] ?
                instance.statistics.slice(0, 3).map(stat => statisticsItem(stat))
                : instance.statistics.map(stat => statisticsItem(stat))}
                <div className="p-2 mt-0 text-center">
                  <i className="fal p-2 fa-calendar" />
                  <p className="small m-0"> Actualizado { moment(instance.date).fromNow() } </p>
                </div>
              </div>
            </Col>
          </Row>
        </Link>
      </Card>
    );

    const imgTrueCard = instance => (
      <Card className="bg-light border-0 rounded-0 small-enlarge two-col">
        <Row noGutters>
          <Link to={`/corrupcion-caso/${instance.id}`}>
            <Col
              xs={layoutMap[instance.size].xs.inner.imgWrapper}
              sm={layoutMap[instance.size].sm.inner.imgWrapper}
              md={layoutMap[instance.size].md.inner.imgWrapper}
              lg={layoutMap[instance.size].lg.inner.imgWrapper}
              xl={layoutMap[instance.size].xl.inner.imgWrapper}
            >
              <img
                className="img-cover"
                src={instance.images[0].image}
                alt="Something"
              />
            </Col>
            <Col
              xs={layoutMap[instance.size].xs.inner.contentWrapper}
              sm={layoutMap[instance.size].sm.inner.contentWrapper}
              md={layoutMap[instance.size].md.inner.contentWrapper}
              lg={layoutMap[instance.size].lg.inner.contentWrapper}
              xl={layoutMap[instance.size].xl.inner.contentWrapper}
            >
              <CardBody className="p-2">
                <h4 className="card-title mb-0">{instance.title}</h4>
              </CardBody>
              <div className="d-flex justify-content-between bg-eb rounded-0">
                {/* If there are three or more statistics,
                only display limits of 3 or map less than that */}
                {instance.statistics && instance.statistics[3] ?
                instance.statistics.slice(0, 3).map(stat => statisticsItem(stat))
                : instance.statistics.map(stat => statisticsItem(stat))}
                <div className="p-2 mt-0 text-center">
                  <i className="fal p-2 fa-calendar" />
                  <p className="small m-0"> Actualizado { moment(instance.date).fromNow() } </p>
                </div>
              </div>
            </Col>
          </Link>
        </Row>
      </Card>
    );

    return (instance.images ? imgTrueCard(instance) : imgFalseCard(instance));
  }
}

export default CorruptionCaseCard;
