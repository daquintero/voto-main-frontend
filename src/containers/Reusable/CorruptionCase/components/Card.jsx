// Generic InformativeSnippet Card
import React, { PureComponent } from 'react';
import { Card, CardBody, CardLink, CardImg } from 'reactstrap';
import { CardPropTypes, CardDefaultPropTypes } from './CorruptionCasePropTypes';

class CorruptionCaseCard extends PureComponent {
  static propTypes = CardPropTypes;
  static defaultProps = CardDefaultPropTypes;
  render() {
    const { info } = this.props;
    return (
      <Card className="bg-white">
        <CardImg
          width="100%"
          src={info.images[0].image}
          alt="Something"
        />
        <CardBody className="p-2">
          <h4 className="card-title">{info.title}</h4>
          <div className="d-flex justify-content-between">
            <CardLink className="p-2 text-center justify-content-center" href={info.statistics[0].link} >
              <i className={`fal p-2 fa-2x fa-${info.statistics[0].icon}`} aria-hidden="true" />
              <div className="justify-content-between">
                <div> {info.statistics[0].data} </div>
                <div> {info.statistics[0].title} </div>
              </div>
            </CardLink>
            {info.statistics[1] ? (
              <CardLink className="p-2 text-center justify-content-center" href={info.statistics[0].link} >
                <i className={`fal p-2 fa-2x fa-${info.statistics[1].icon}`} aria-hidden="true" />
                <div className="justify-content-between">
                  <div> {info.statistics[1].data} </div>
                  <div> {info.statistics[1].title} </div>
                </div>
              </CardLink>
            ) : null}
            {info.statistics[2] ? (
              <CardLink className="p-2 text-center justify-content-center" href={info.statistics[0].link} >
                <i className={`fal p-2 fa-2x fa-${info.statistics[2].icon}`} aria-hidden="true" />
                <div className="justify-content-between">
                  <div> {info.statistics[2].data} </div>
                  <div> {info.statistics[2].title} </div>
                </div>
              </CardLink>
            ) : null}
          </div>
        </CardBody>
      </Card>
    );
  }
}

export default CorruptionCaseCard;
