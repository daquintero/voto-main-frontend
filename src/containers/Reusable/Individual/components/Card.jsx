// Generic Individual Card
import React, { PureComponent } from 'react';
import { Card, CardBody, Row, Col } from 'reactstrap';
// Routing for each card
import { Link } from 'react-router-dom';
import { CardPropTypes, CardDefaultPropTypes } from './IndividualPropTypes';
// Numerical Formatting
// import numeral from 'numeral'; TODO FINISH
// Layout Map
import layoutMap from '../styling/layoutMap';

class IndividualCard extends PureComponent {
  static propTypes = CardPropTypes;
  static defaultProps = CardDefaultPropTypes;

  render() {
    const { objIn } = this.props;

    const imgFalseCard = obj => (
      <Card className="bg-white border-0 h-100 p-2">
        <Row noGutters className="h-100">
          <Col
            xs={layoutMap[obj.size].xs.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
            sm={layoutMap[obj.size].sm.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
            md={layoutMap[obj.size].md.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
            lg={layoutMap[obj.size].lg.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
            xl={layoutMap[obj.size].xl.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
          >
            <CardBody className="p-2">
              <Row noGutters>
                <Col
                  xs={layoutMap[obj.size].xs.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
                  sm={layoutMap[obj.size].sm.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
                  md={layoutMap[obj.size].md.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
                  lg={layoutMap[obj.size].lg.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
                  xl={layoutMap[obj.size].xl.inner.imgWrapper + layoutMap[obj.size].xs.inner.contentWrapper}
                  className="text-center p-1"
                >
                  <h4 className="card-title mb-0">{obj.name}</h4>
                  <p className="font-italic mt-0">{obj.alias}</p>
                  <Row noGutters className="emp-card-info-btm">
                    <Col xs={4}>
                      <i
                        className={`text-center fal ${obj.corruptionIndex && obj.corruptionIndex.change ?
                          'fa-caret-up text-success fa-2x' :
                          'fa-caret-down text-danger fa-2x'}`}
                      />
                      <p className="m-0 text-center">{obj.corruptionIndex.value}</p>
                      <p className="m-0 small">Indíce</p>
                    </Col>
                  </Row>
                </Col>
              </Row>
            </CardBody>
          </Col>
        </Row>
      </Card>
    );

    const imgTrueCard = obj => (
      <Card className="bg-white border-0">
        <Row noGutters>
          <Col
            xs={layoutMap[obj.size].xs.inner.imgWrapper}
            sm={layoutMap[obj.size].sm.inner.imgWrapper}
            md={layoutMap[obj.size].md.inner.imgWrapper}
            lg={layoutMap[obj.size].lg.inner.imgWrapper}
            xl={layoutMap[obj.size].xl.inner.imgWrapper}
          >
            <img
              className="img-cover"
              src={obj.images[0].image}
              alt="Something"
            />
          </Col>
          <Col
            xs={layoutMap[obj.size].xs.inner.contentWrapper}
            sm={layoutMap[obj.size].sm.inner.contentWrapper}
            md={layoutMap[obj.size].md.inner.contentWrapper}
            lg={layoutMap[obj.size].lg.inner.contentWrapper}
            xl={layoutMap[obj.size].xl.inner.contentWrapper}
          >
            <CardBody className="p-2">
              <Row noGutters>
                <Col xs={3} className="text-center p-1">
                  <i
                    className={`text-center fal ${obj.corruptionIndex && obj.corruptionIndex.change ?
                      'fa-caret-up text-success fa-2x' :
                      'fa-caret-down text-danger fa-2x'}`}
                  />
                  <p className="m-0 text-center">{obj.corruptionIndex.value}</p>
                  <p className="m-0 small">Indíce</p>
                </Col>
                <Col xs={9} className="mb-2">
                  <h4 className="card-title mb-0 text-right">{obj.name}</h4>
                  <p className="text-right mt-0 font-italic">{obj.alias}</p>
                </Col>
              </Row>
            </CardBody>
          </Col>
        </Row>
      </Card>
    );

    return (
      <Link to={`/individuo/${objIn.id}`}>
        { objIn.images ? imgTrueCard(objIn) : imgFalseCard(objIn) }
      </Link>
    );
  }
}

export default IndividualCard;
