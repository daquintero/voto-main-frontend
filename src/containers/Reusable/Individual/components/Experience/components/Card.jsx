// Experience Card
// Libraries
import React, { Component } from 'react';
import { Card, CardHeader, CardBody } from 'reactstrap';

// Prop Types
import { CardPropTypes } from './ControversyPropTypes';

// Declaration
class ContCard extends Component {
  static propTypes = CardPropTypes;
  constructor(props) {
    super(props);
    this.state = {
      open: false,
    };
  }
  openMore = () => {
    this.setState(prevState => ({ open: !prevState.open }));
  };
  render() {
    const { objIn } = this.props;
    const { open } = this.state;
    return (
      <Card className="bg-light border-0 rounded-0 small-enlarge" onClick={this.openMore}>
        <CardHeader className="pt-1 pb-1 ">
          <h6 className=" p-1 mb-0"><i className="fa p-1 fa-university" />{objIn.title}</h6>
          <p className="small">{objIn.organization}</p>
        </CardHeader>
        { open && objIn.description ? (
          <CardBody className="p-2">
            <p className="p-1 m-0">{objIn.description}</p>
          </CardBody>
        ) : null }
      </Card>
    );
  }
}

export default ContCard;
