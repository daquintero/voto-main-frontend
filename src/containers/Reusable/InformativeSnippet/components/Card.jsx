// Generic InformativeSnippet Card
import React, { PureComponent } from 'react';
import { Card, CardBody, CardLink, CardImg, Row, Col } from 'reactstrap';
import moment from 'moment'; // Date conversion
import 'moment/locale/es'; // Date language
import { CardPropTypes, CardDefaultPropTypes } from './InformativeSnippetPropTypes';

// Configure moment settings
moment.locale('es');

class InformativeSnippetCard extends PureComponent {
  static propTypes = CardPropTypes;
  static defaultProps = CardDefaultPropTypes;
  render() {
    const { info } = this.props;
    return (
      <Card className="bg-white">
        <Row noGutters>
          <Col xs={4} sm={6}>
            <CardImg
              width="100%"
              src={info.images[0].image}
              alt="Something"
            />
          </Col>
          <Col xs={8} sm={6}>
            <CardBody className="p-2">
              <h4 className="card-title">{info.title}</h4>
              <div className="d-flex justify-content-between">
                <CardLink className="p-2 text-center justify-content-center" href={info.statistics[0].link} >
                  <i className={`fa p-2 fa-2x fa-${info.statistics[0].icon}`} aria-hidden="true" />
                  <div className="justify-content-between">
                    <div> {info.statistics[0].data} </div>
                    <div> {info.statistics[0].title} </div>
                  </div>
                </CardLink>
                <div className="p-2 mt-0 text-center">
                  <i className="fa p-2 fa-calendar fa-2x" />
                  <p> Publicado { moment(info.date).fromNow() } </p>
                </div>
              </div>
            </CardBody>
          </Col>
        </Row>
      </Card>
    );
  }
}

export default InformativeSnippetCard;
