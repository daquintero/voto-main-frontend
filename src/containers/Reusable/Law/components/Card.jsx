// Law Card

// Libraries
import React, { PureComponent } from 'react';
import { ListGroupItem, Col, Row } from 'reactstrap';

// Props
import { lawCardPropTypes } from './LawPropTypes';


class LawsList extends PureComponent {
  static propTypes = lawCardPropTypes;

  render() {
    const { law } = this.props;
    return (
      <ListGroupItem className="list-right" tag="a" href={law.source}>
        <Row noGutters>
          <Col xs={2} className="align-middle">
            <i className="fal fa-gavel" />
          </Col>
          <Col xs={2} >
            <p className="small m-0 p-0"> No. </p>
            <h4 className="p-1">{law.code}</h4>
          </Col>
          <Col xs={8} className="pl-3 ">
            <p className="small text-left ">{law.briefDescription}</p>
          </Col>
        </Row>
      </ListGroupItem>
    );
  }
}

export default LawsList;
