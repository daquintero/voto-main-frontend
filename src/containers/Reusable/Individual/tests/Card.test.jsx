// Test for the Home CorruptionCase Component
import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../components/Card';

describe('<Card />', () => {
  it('HomePage renders correctly without crashing', () => {
    const tree = renderer
      .create(<Card />)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
