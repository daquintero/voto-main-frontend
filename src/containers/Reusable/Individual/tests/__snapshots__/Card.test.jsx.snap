// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> HomePage renders correctly without crashing 1`] = `
<div
  className="bg-white card"
>
  <img
    alt="Something"
    className="card-img"
    src=""
    width="100%"
  />
  <div
    className="p-2 card-body"
  >
    <h4
      className="card-title"
    >
      
    </h4>
  </div>
</div>
`;
